tests/spec/test_struct.py

```diff
@@ -491,3 +491,24 @@ def test_has_thrift_module(loads):
         }
     ''')
     assert module is module.Foo.__thrift_module__
+
+
+def test_has_implemented_hash(loads):
+    module = loads('''
+        struct Foo {
+            1: required string a
+        }
+    ''')
+    assert hasattr(module.Foo, '__hash__')
+
+
+def test_hash_ignores_non_hashable_types(loads):
```

Comment:
I think these tests can be combined with the union test cases.

```python
@pytest.mark.parametrize('name, src, kwargs', [
    ('Foo', 'struct Foo { 1: required string a }', {'a': 'zzz'}),
    (
        'Bar', '''
            struct Bar {
                1: required string a
                2: required list<string> b
                3: required map<string, string> c = {}
            }
        ''',
        {'a': 'zzz', 'b': ['bar']},
    ),
    # ...
])
def test_hashable(loads, name, src, kwargs):
    module = loads(src)
    klass = getattr(module, name)
    obj = klass(**kwargs)
    assert hash(obj)
```

This also easily accounts for future test cases.

(We'll want to put this in a `tests/spec/test_hashable.py` instead of union or
struct-specific tests.)